### ISSUE TITLE
Is there a means of forcing a re-render of a single component outside of dereference?

### ISSUE THREAD

**Issue by kvey** [reagent-project/reagent#123]
Dereference (or updating of an r/atom that is deferenced inside of a component)

**Comment by mike-thompson-day8** [comment]
I'm not entirely clear on your question but ...

There are 2 things which will trigger (force?) the rerender of component:
  - a change (reset!) to a `reagent/atom` dereferenced within a component
  - a change in the component's props (parameters)

An entire hierarchy of components can also be "forced" to reender via either:
   - `reagent/force-update-all`  (but be careful of #95)
   - completely remounting the root component (some explanation in  #112)

**Comment by mike-thompson-day8** [comment]
@kvey can this be closed?

**Comment by kvey** [comment]
I was looking for a means of forcing a re-render of an individual
component. This can be closed. I was attempting to write a transitiongroup
without using react's lifecycle methods.

**Comment by mike-thompson-day8** [comment]
There is this ugly trick:
```
(def   forcer  (reagent/atom 0))

(defn my-component
   []
   @forcer
   [:div  something])
```

When you want to force a rerender of `my-component` you can just:
`(swap! forcer  inc)`

You will need to close this issue, I don't have perms.

**Comment by dchacke** [comment]
There is [`(reagent.core/force-update this)`](http://reagent-project.github.io/docs/master/reagent.core.html#var-force-update).

I just used it to add an `.active` class to navigation elements to highlight them if they correspond to the current route, for which I needed to listen to `window.onhashchange`. Updating an atom on hash change and dereferencing it in the navigation element's component seemed like a lot of unnecessary overhead.
